rfcs/20190116-embedding-partitioned-variable.md

```diff
@@ -0,0 +1,498 @@
+# Embedding and Partitioned Variable in TF 2.0
+
+
+| Status        | Proposed                                                                  |
+| :------------ | :------------------------------------------------------------------------ |
+| **Author** | Yuefeng Zhou (yuefengz@google.com)                                        |
+|**Sponsors**| Alexandre Passos (apassos@google.com), Martin Wicke (wicke@google.com), Josh Levenberg  (joshl@google.com) |
+|**Updated** | 2019-1-16                                                                 |
+
+
+Summary: this RFC describes the design and interfaces for embeddings and partitioned variables in TF 2.0 based on Distribution Strategy. We propose to support embeddings and sharded layers with DistributionStrategy and Keras' layer API. We also propose to shard embeddings and layers only in `ParameterServerStrategy` and mirror them in `MirroredStrategy` or `CollectiveAllReduceStrategy.` 
+
+
+## Background
+
+
+### Embeddings
+
+
+An [embedding](https://www.tensorflow.org/guide/embedding) is a mapping from discrete objects, such as words, to vectors of real numbers. In TensorFlow implementation, we use a variable to hold all these vectors for a feature column. All vectors in an embedding variable have the same size but the sizes of embedding variables may vary. 
+
+We don't always use the whole embedding variable in a training step, the result of embedding lookup is <code>[IndexedSlices](https://www.tensorflow.org/api_docs/python/tf/IndexedSlices)</code> which is essentially a dict mapping from indices to vectors. So is the gradient to an embedding variable.
```

Comment:
The result of embedding lookup is not indexedslices; the gradient of embedding lookup is indexedslices

Reply:
Done.